src/modules/commander/commander.cpp

```diff
@@ -3618,6 +3618,12 @@ set_control_mode()
 		control_mode.flag_control_altitude_enabled = (!offboard_control_mode.ignore_velocity ||
 				!offboard_control_mode.ignore_position) && !control_mode.flag_control_acceleration_enabled;
 
+		// These are all required for altitude control
+		control_mode.flag_control_altitude_enabled	= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_altitude_enabled;
+		control_mode.flag_control_attitude_enabled	= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_attitude_enabled;
+		control_mode.flag_control_rates_enabled		= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_rates_enabled;
+		control_mode.flag_control_climb_rate_enabled= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_climb_rate_enabled;
+
 		break;
```

Comment:
I started to so something similar, but my intention was to make it even more general such that position/velocity can be mixed with altitude/climbrate. the branch is [here](https://github.com/Stifael/Firmware/tree/offboard_enable_feedforward), but I don't want to open a PR because I want to get Flightask in first. 

Anyway, I also had to change how to commander handles the offboard control modes, and I came to the conclusion that the legacy mapping from `offbboard_control_mode` to `control_mode` is not correct. For instance, the mapping to `flag_control_altitude_enabled` is wrong because neither velocity nor position control is required. Hence, instead of just adding those new lines, I would fix the previous settings (see [here](https://github.com/Stifael/Firmware/blob/offboard_enable_feedforward/src/modules/commander/commander.cpp#L3593-L3624))

Reply:
@Stifael What is the progress on your offboard refactor? Have you already made a PR?